src/client/power-pages/actions-hub/ActionsHubTreeDataProvider.ts

```diff
@@ -89,6 +90,32 @@ export class ActionsHubTreeDataProvider implements vscode.TreeDataProvider<Actio
                 } catch (error) {
                     oneDSLoggerWrapper.getLogger().traceError(Constants.EventNames.ACTIONS_HUB_REFRESH_FAILED, error as string, error as Error, { methodName: this.refresh.name }, {});
                 }
+            }),
+
+            vscode.commands.registerCommand("powerpages.actionsHub.switchEnvironment", async () => {
+                const authInfo = pacAuthManager.getAuthInfo();
+                if(authInfo) {
+                    const envListOutput = await pacWrapper.orgList();
```

Comment:
Can we preload this during initialization and refresh when refreshing the environment?

Reply:
This list is linked to auth, and I think refreshing it on every environment switch is an expensive operation and not needed also. We can fetch it during auth change but it will become stale if later a new env is added on the DV side. So, I believe fetching it on the user's trigger is good.